Fixing camelCaseKeys export

GreenGremlin opened the pull request:
Currently, the `camelCaseKeys` is an object with a `camelCaseKeys` property. Which means it has to be accessed as `camelCaseKeys.camelCaseKeys`:

```js
import { camelCaseKeys } from 'launchdarkly-react-client-sdk';

const flags = camelCaseKeys.camelCaseKeys(ldClient.allFlags());
```

Or the import can be renamed:

```js
import { camelCaseKeys as LDUtils } from 'launchdarkly-react-client-sdk';

const flags = LDUtils.camelCaseKeys(ldClient.allFlags());
```

This change fixes the export so that the `camelCaseKeys` function is directly exported, which seems to be what was intended by the existing code.

As it is, this is a breaking change, but I can update it to be backwards compatible if needed.

**Requirements**

- [ ] I have added test coverage for new or changed functionality **This PR does not include new functionality**
- [x] I have followed the repository's [pull request submission guidelines](../blob/master/CONTRIBUTING.md#submitting-pull-requests)
- [x] I have validated my changes against all supported platform versions

**Related issues**

Provide links to any issues in this repository or elsewhere relating to this pull request.

**Describe the solution you've provided**

Provide a clear and concise description of what you expect to happen.

**Describe alternatives you've considered**

Provide a clear and concise description of any alternative solutions or features you've considered.

**Additional context**

Add any other context about the pull request here.

bwoskow-ld commented:
Hi @GreenGremlin ,

Thanks for pointing this out. While relatively minor, you're right -- we didn't intend for consumers to have to call `camelCaseKeys.camelCaseKeys(flags)` as that's unnecessarily redundant.

> As it is, this is a breaking change, but I can update it to be backwards compatible if needed.

We definitely don't want to ship a breaking change. We're not ready to release a major version of the SDK at this time. With this in mind, please update your PR so that this is a non-breaking change.

Thanks,
Ben

GreenGremlin commented:
> Hi @GreenGremlin ,
> 
> Thanks for pointing this out. While relatively minor, you're right -- we didn't intend for consumers to have to call `camelCaseKeys.camelCaseKeys(flags)` as that's unnecessarily redundant.
> 
> > As it is, this is a breaking change, but I can update it to be backwards compatible if needed.
> 
> We definitely don't want to ship a breaking change. We're not ready to release a major version of the SDK at this time. With this in mind, please update your PR so that this is a non-breaking change.
> 
> Thanks,
> Ben

Updated. I added a `TODO` comment. Let me know if you have a preferred format for TODOs.

bwoskow-ld commented on src/utils.ts:
```diff
@@ -20,4 +20,7 @@ export const camelCaseKeys = (rawFlags: LDFlagSet) => {
   return flags;
 };
 
+// TODO: Remove this for the next major version. This was added to maintain backwards compatibility.
```
Rather than using a TODO like this, I'd prefer adding a JSDoc with a `@deprecated` tag. Once that's in I can generate the docs locally and make sure it looks acceptable.

[Here](https://github.com/launchdarkly/js-sdk-common/blob/master/typings.d.ts#L216) is some language from an internal dependency which you can copy here.

GreenGremlin replied:
Updated! Sorry for the delay

GreenGremlin replied:
It looks like the `@deprecated` tag is deprecating the `camelCaseKeys` function entirely. Any idea how to just deprecate the `camelCaseKeys. camelCaseKeys` property?

GreenGremlin replied:
I think I figured it out. I had to disable the tslint deprecation rule for that line. I verified that using `camelCaseKeys.camelCaseKeys` now triggers the error, but using `camelCaseKeys` directly does not.

bwoskow-ld commented:
FYI - while this PR remained under review, we added another utility function. Please update your changes accordingly to resolve the conflict.

GreenGremlin commented:
> FYI - while this PR remained under review, we added another utility function. Please update your changes accordingly to resolve the conflict.

I've rebased the latest changes from the contrib branch.

bwoskow-ld commented:
Thanks for the contribution @GreenGremlin ! We'll release this as part of the next React SDK release.
